tests/cloudlaunchprocess/test.csv

```diff
@@ -0,0 +1,27 @@
+action,ms,bytes,lang,code,repeat,minver,comment
+comment,0,,"Test 1 - Launch and then kill a process with all possible cmd line args"
+before,0,0,q,discport:1+"I"$getenv[`KDBBASEPORT] ,,"Find the port of the discovery process"
+run,0,0,q,launch input1,,"Start a new process with extra command line args"
+run,0,0,q,system "sleep 3",,"Wait for process to start"
+run,0,0,q,h:hopen `::7124:admin:admin ,,"Attempt to connect to the new process"
+true,0,0,q,`default=h".wdb.writedownmode",,"Check that connection has been made"
+true,0,0,q,110011011b~(2 _ (2 cut h".z.X")[;1]) in value input1,,"Check that the cmd line args are as expected "
+run,0,0,q,pid:"I"$-2 _ raze raze string a where "q" in' a:.proc.sys "pgrep -lf wdb1",,"Find the PID of the launched process"
+run,0,0,q,kill "wdb1",,"Kill the process"
+true,0,0,q,10h=type @[{h x};".wdb.writedownmode";{"Error: failure to connect"}],,"Check that the handle to the process is no longer open"
+true,0,0,q,not first "q" in' b:.proc.sys "pgrep -lf wdb1",,"Check that the PID of the old process is no longer in use"
+
+comment,0,,"Test 2 - Launch and then kill a process with no extra cmd line args, to test defaulting"
+run,0,0,q,launch input2,,,"Start a new process with no extra command line args, hence relying on defaulting in cloudutils.q"
+run,0,0,q,.proc.sys "sleep 2",,,"Wait for process to start"
+run,0,0,q,"h2:hopen `$""::"",string[discport],"":admin:admin""",,,"Connect to discovery1"
+run,0,0,q,"proc_port:""I""$(neg count string pid)#.Q.s1 h2""exec hpup from .servers.SERVERS where not null w,proctype=`wdb""",,,"Find the process port via discovery"
```

Comment:
I think here you can use `.servers.querydiscovery` https://github.com/AquaQAnalytics/TorQ/blob/6cec4609ca7931983b527987202d9855df98ba87/code/handlers/trackservers.q#L205

So something like

```suggestion
run,0,0,q,proc:.servers.querydiscovery`test,,,"Find the process port via discovery"
```

this will return a table of all processes discovery knows about with proctype `` `test`` - you can then do e.g. ``exec hpup from proc where procname=`test3`` to get the hpup for the proc in question

I think your code to extract just the port number is also slightly flawed (pid could be a different length to port number) and unnecessary - see comment on line below